src/thunderbots/software/test/ai/hl/stp/action/chip_action.cpp

```diff
@@ -7,164 +7,228 @@
-                        AngularVelocity::zero(), Timestamp::fromSeconds(0));
+    Robot robot(0, Point(-0.3, 0), Vector(), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Ball ball({0, 0}, robot.position(), Timestamp::fromSeconds(0));
     ChipAction action = ChipAction();
 
-    auto intent_ptr =
-        action.updateStateAndGetNextIntent(robot, Point(0, 0), Angle::zero(), 5.0);
+    auto intent_ptr = action.updateStateAndGetNextIntent(robot, ball, ball.position(),
+                                                         Angle::zero(), 5.0);
 
     // Check an intent was returned (the pointer is not null)
     EXPECT_TRUE(intent_ptr);
     EXPECT_FALSE(action.done());
 
-    ChipIntent chip_intent = dynamic_cast<ChipIntent &>(*intent_ptr);
-    EXPECT_EQ(0, chip_intent.getRobotId());
-    EXPECT_EQ(Point(0, 0), chip_intent.getChipOrigin());
-    EXPECT_EQ(Angle::zero(), chip_intent.getChipDirection());
-    EXPECT_EQ(5.0, chip_intent.getChipDistance());
+    try
+    {
+        ChipIntent kick_intent = dynamic_cast<ChipIntent &>(*intent_ptr);
+        EXPECT_EQ(0, kick_intent.getRobotId());
+        EXPECT_EQ(Point(0, 0), kick_intent.getChipOrigin());
+        EXPECT_EQ(Angle::zero(), kick_intent.getChipDirection());
+        EXPECT_EQ(5.0, kick_intent.getChipDistance());
+    }
+    catch (...)
+    {
+        ADD_FAILURE() << "Chip intent not returned by Chip Action!";
+    }
 }
 
 TEST(ChipActionTest, robot_behind_ball_chipping_towards_negative_x_positive_y)
 {
-    Robot robot       = Robot(0, Point(-2.3, 2.1), Vector(), Angle::quarter(),
-                        AngularVelocity::zero(), Timestamp::fromSeconds(0));
+    Robot robot(0, Point(-2.4, 2.3), Vector(), Angle::quarter(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Ball ball({-2.5, 2.5}, robot.position(), Timestamp::fromSeconds(0));
     ChipAction action = ChipAction();
 
-    auto intent_ptr = action.updateStateAndGetNextIntent(robot, Point(-2.5, 2.5),
+    auto intent_ptr = action.updateStateAndGetNextIntent(robot, ball, ball.position(),
                                                          Angle::ofDegrees(105), 5.0);
 
     // Check an intent was returned (the pointer is not null)
     EXPECT_TRUE(intent_ptr);
     EXPECT_FALSE(action.done());
 
-    ChipIntent chip_intent = dynamic_cast<ChipIntent &>(*intent_ptr);
-    EXPECT_EQ(0, chip_intent.getRobotId());
-    EXPECT_EQ(Point(-2.5, 2.5), chip_intent.getChipOrigin());
-    EXPECT_EQ(Angle::ofDegrees(105), chip_intent.getChipDirection());
-    EXPECT_EQ(5.0, chip_intent.getChipDistance());
+    try
+    {
+        ChipIntent kick_intent = dynamic_cast<ChipIntent &>(*intent_ptr);
+        EXPECT_EQ(0, kick_intent.getRobotId());
+        EXPECT_EQ(Point(-2.5, 2.5), kick_intent.getChipOrigin());
+        EXPECT_EQ(Angle::ofDegrees(105), kick_intent.getChipDirection());
+        EXPECT_EQ(5.0, kick_intent.getChipDistance());
+    }
+    catch (...)
```

Comment:
Could we be a bit more specific in what exceptions we're catching here? We're really only looking for `std::bad_cast` right? Also if so, this try-catch should probably only be around the cast itself.

Same goes for the other instances of this below.

Reply:
This is consistent with how we do it in other tests, but I do agree. This isn't a blocking issue for this PR so I'd prefer to leave it and address all the test casts in a separate PR.